### PR TITLE
Add startupProbe to containers created by the operator

### PR DESCRIPTION
**Pull request by jhunzik**
## About
Prior to this change, a container created by the KEIP operator only had 40 seconds to become healthy before being killed. In some scenarios where many routes are deployed this resulted in container boot-looping. To fix this each container is now configured to use a startupProbe to give the container up to 120 seconds to registry as healthy before the liveness and readiness probes to take over.

## Hero Instructions
1. Start with a fresh Minikube cluster or equivalent
2. From the `operator/webhook` directory: `make build && minikube image load ghcr.io/octoconsulting/keip/route-webhook:0.6.1`
3. In the operator directory, run `make all` to deploy the integration route controller.
4. Deploy the example route: `kubectl apply -k operator/example`
5. Port forward the pod and verify it can be hit in the browser: `kubectl port-forward testroute-74d574bf85-tbv9m 8443:8443`
6. Verify the pod has a startupProbe configured: `kubectl describe <pod>`

### CODE REVIEW

File: operator/webhook/introute/sync.py [hunk 1]
```diff
@@ -291,15 +291,29 @@ def _create_pod_template(parent, labels, integration_image):
                             "port": management_port,
                             "scheme": scheme
                         },
-                        "initialDelaySeconds": 10,
+                        "failureThreshold": 3,
+                        "periodSeconds": 5,
```
Comment (pelletier2017) [inline operator/webhook/introute/sync.py:295]: Could we bump up period seconds to 10 or more? The default is 10 and I don't think we need to check quite as often.

File: operator/webhook/introute/sync.py [hunk 2]
```diff
@@ -291,15 +291,29 @@ def _create_pod_template(parent, labels, integration_image):
...
                     },
                     "readinessProbe": {
                         "httpGet": {
                             "path": "/actuator/health/readiness",
                             "port": management_port,
                             "scheme": scheme
                         },
-                        "initialDelaySeconds": 10,
+                        "failureThreshold": 1,
+                        "periodSeconds": 5,
```
Comment (pelletier2017) [inline operator/webhook/introute/sync.py:305]: Same question for readiness probe periodSeconds.

File: operator/webhook/introute/sync.py [hunk 3]
```diff
@@ -291,15 +291,29 @@ def _create_pod_template(parent, labels, integration_image):
...
+                        "timeoutSeconds": 3
                     },
                     "readinessProbe": {
                         "httpGet": {
                             "path": "/actuator/health/readiness",
                             "port": management_port,
                             "scheme": scheme
                         },
-                        "initialDelaySeconds": 10,
+                        "failureThreshold": 1,
```
Comment (pelletier2017) [inline operator/webhook/introute/sync.py:304]: Do we want it to fail 3x for failure threshold? I figure you may want readiness checks to be more aggressive than liveness but I don't know if we want to keep it at just 1.

Comment (jhunzik) [reply operator/webhook/introute/sync.py:304]: I think it would be better to keep it at 1 to minimize the change a service routes 
traffic to a pod that can't accept it.

Comment (jhunzik) [reply operator/webhook/introute/sync.py:303]: I could also see having at just 1 lead to some false positives. I'll update it to 2

**Review by pelletier2017 — APPROVED** [review]

File: operator/Makefile [hunk 4]
```diff
@@ -22,10 +22,16 @@ prep-release:
...
 	ls -al ./output
 
-metacontroller/deploy:
+cert-manager/deploy:
+	$(KUBECTL) apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.0/cert-manager.yaml
+
+cert-manager/undeploy:
+	-$(KUBECTL_DELETE) apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.0/cert-manager.yaml
+
+metacontroller/deploy: cert-manager/deploy
```
Comment (a-asaad) [inline operator/Makefile:31]: Should the metacontroller have a target dependency on the cert-manager? I don't think the cert-manager is required to deploy the metacontroller.

File: operator/Makefile [hunk 5]
```diff
@@ -22,10 +22,16 @@ prep-release:
 	kustomize build ./crd > ./output/crd.yaml
 	ls -al ./output
 
-metacontroller/deploy:
+cert-manager/deploy:
```
Comment (a-asaad) [inline operator/Makefile:25]: :question: Why was a `cert-manager` target added to the Makefile? It seems out of scope for deploying the Keip operator, since neither the metacontroller nor the keip webhook depend on it explicitly.

**Comment by a-asaad** [comment]
Should we also bump the top-level operator version in https://github.com/OctoConsulting/keip/blob/main/operator/Makefile

File: operator/Makefile (hunk 5 above)
Comment (jhunzik) [reply operator/Makefile:25]: The example route used TLS so cert-manager was needed to run it. Maybe that should just be entirely optional and I'll add back the readme steps to install cert-manager rather than doing it here.

**Comment by jhunzik** [comment]
> Should we also bump the top-level operator version in https://github.com/OctoConsulting/keip/blob/main/operator/Makefile?

Good point, I knew I was forgetting something.

File: operator/Makefile (hunk 5 above)
Comment (a-asaad) [reply operator/Makefile:25]: Yeah, I think that makes more sense, since technically the cert could from anywhere, so the use of cert-manager to create it is specific to this example.
